python-checks/src/test/resources/checks/identicalExpressionOnBinaryOperator.py

```diff
@@ -41,3 +41,21 @@ def work():
 exclusion = 1 << 1
 exclusion2 = (a * b) << 1
 result = x @ x #compliant matrix operator should not raise issue on this rule.
+
+def foo(): ...
+
+class MyClass:
+  def bar(): ...
+
+def no_issues_on_function_calls():
+  if foo() == foo(): ...
+  if foo().x == foo().x: ...
+  if MyClass() == MyClass(): ...
+  my_class = MyClass()
+  if my_class.bar() == my_class.bar(): ...
+
+def no_issues_within_try_except():
+  try:
+    foo(c)
+  except ValueError:
+    return c / c
```

Comment:
why do we want to exclude this one?

Reply:
I didn't keep the comment for this one and simplified a bit the example, but it's an actual case where they want to assess the value of parameter `c`: `0/0` which would raise an exception `inf/inf` which would return `nan` and any other number which would return `1`.
I'm not sure if there are other legitimate uses for this construct, but I prefer to avoid being noisy in try/except blocks for that reason.

Reply:
ok..as discussed please create a separate ticket for the try/except FP